include/sway/layers.h

```diff
@@ -47,6 +51,7 @@ struct sway_layer_subsurface {
 	struct wl_listener unmap;
 	struct wl_listener destroy;
 	struct wl_listener commit;
+	struct wl_listener surface_destroy;
```

Comment:
Instead of setting up an event listener, can we just iterate over the layer sub-surfaces and destroy them in `handle_destroy`?

Reply:
Probably. I did it this way only because everything else here seems to work this way. What you suggest is probably better.

Reply:
The rest of the event listeners are listening to wlroots events. wlroots doesn't know about Sway, so we need this complexity in the wlroots API.

However for Sway-internal stuff we shouldn't need it.